cfn-resources/private-endpoint/cmd/resource/resource.go

```diff
@@ -79,111 +67,130 @@ func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler
+				Message:         "Create Success",
+				ResourceModel:   currentModel,
+			}, nil
 		}
 
-		privateEndpointInput := make([]privateendpoint.AtlasPrivateEndpointInput, len(awsPrivateEndpointOutput))
+		status := *privateEndpoint.ConnectionStatus
+		return handler.ProgressEvent{
+			OperationStatus:      handler.InProgress,
+			Message:              "Create in progress",
+			ResourceModel:        currentModel,
+			CallbackDelaySeconds: 20,
+			CallbackContext: map[string]interface{}{
+				"state": status,
+			}}, nil
+	}
 
-		for i, awsPe := range awsPrivateEndpointOutput {
-			privateEndpointInput[i] = privateendpoint.AtlasPrivateEndpointInput{
-				VpcID:               awsPe.VpcID,
-				SubnetIDs:           awsPe.SubnetIDs,
-				InterfaceEndpointID: awsPe.InterfaceEndpointID,
-			}
-		}
+	endpointRequest := admin.CreateEndpointRequest{
+		Id: currentModel.InterfaceEndpointId,
+	}
 
-		pe := privateendpoint.Create(mongodbClient, *currentModel.GroupId, privateEndpointInput, peConnection.ID)
+	privateEndpointRequest := client.AtlasV2.PrivateEndpointServicesApi.CreatePrivateEndpoint(context.Background(), *currentModel.ProjectId,
+		CloudProvider, *currentModel.EndpointServiceId, &endpointRequest)
 
-		return addModelToProgressEvent(&pe, currentModel), nil
-	default:
-		ValidationOutput, progressEvent := privateendpoint.ValidateCreationCompletion(mongodbClient, *currentModel.GroupId, req)
-		if progressEvent != nil {
-			return addModelToProgressEvent(progressEvent, currentModel), nil
-		}
+	_, response, err := privateEndpointRequest.Execute()
+	defer response.Body.Close()
+	if err != nil {
+		return progress_events.GetFailedEventByResponse(fmt.Sprintf("error creating Serverless Private Endpoint %s",
+				err.Error()), response),
+			nil
+	}
 
-		currentModel.Id = &ValidationOutput.ID
+	currentModel.setPrimaryIdentifier()
 
-		for _, cmpe := range currentModel.PrivateEndpoints {
-			for i := range ValidationOutput.Endpoints {
-				vpe := ValidationOutput.Endpoints[i]
-				if vpe.VpcID == *cmpe.VpcId && CompareSlices(vpe.SubnetIDs, cmpe.SubnetIds) {
-					currentModel.PrivateEndpoints[i].InterfaceEndpointId = &vpe.InterfaceEndpointID
-				}
-			}
-		}
+	log.Print(*currentModel.Id)
+	return handler.ProgressEvent{
+		OperationStatus:      handler.InProgress,
+		Message:              "Create in progress",
+		ResourceModel:        currentModel,
+		CallbackDelaySeconds: 10,
+		CallbackContext: map[string]interface{}{
+			"state": "Pending",
+		}}, nil
+}
 
-		privateEndpointResponse, response, err := mongodbClient.PrivateEndpoints.Get(context.Background(), *currentModel.GroupId, providerName, *currentModel.Id)
-		if err != nil {
-			return progress_events.GetFailedEventByResponse(fmt.Sprintf("Error getting resource : %s", err.Error()),
-				response.Response), nil
-		}
-		currentModel.EndpointServiceName = &privateEndpointResponse.EndpointServiceName
+func (m *Model) setPrimaryIdentifier() {
+	id := *m.InterfaceEndpointId
+	m.Id = &id
```

Comment:
Is there a reason why the private endpoint Model has an Id property that will be the same as InterfaceEndpointId?